src/chains/ethereum/ethereum/tests/provider.test.ts

```diff
@@ -30,6 +30,154 @@ describe("provider", () => {
+      const provider = await getProvider();
+      const options = provider.getOptions();
+      assert(options.miner.timestampIncrement, "clock");
+
+      const accounts = await provider.send("eth_accounts");
+      const tx = { from: accounts[0], gas: "0xfffff" };
+      const sentAfter = Math.floor(Date.now() / 1000);
+      const hash = await provider.request({
+        method: "eth_sendTransaction",
+        params: [tx]
+      });
+      const receipt = await provider.request({
+        method: "eth_getTransactionReceipt",
+        params: [hash]
+      });
+      assert.notStrictEqual(receipt, null);
+      const sentBefore = Math.floor(Date.now() / 1000);
+      const block = await provider.request({
+        method: "eth_getBlockByNumber",
+        params: [receipt.blockNumber, false]
+      });
+      // the `block.timestamp` can be the same as `sentAfter` and/or
+      // `sentBefore` because the precision of `block.timestamp` is 1 second
+      // (floored), and mining happens much quicker than 1 second.
+      assert(
+        parseInt(block.timestamp) >= sentAfter,
+        `block wasn't mined at the right time, should have been on or after ${sentAfter}, was ${parseInt(
+          block.timestamp
+        )}`
+      );
+      assert(
+        parseInt(block.timestamp) <= sentBefore,
+        `block wasn't mined at the right time, should have been on or before ${sentAfter}, was ${parseInt(
+          block.timestamp
+        )}`
+      );
+      await provider.disconnect();
+    });
+
+    it("uses the timestampIncrement option", async () => {
+      const time = new Date("2019-01-01T00:00:00.000Z");
+      const timestampIncrement = 5;
+      const provider = await getProvider({
+        chain: { time },
+        miner: { timestampIncrement }
+      });
+      const accounts = await provider.send("eth_accounts");
+      const tx = { from: accounts[0], gas: "0xfffff" };
+      const hash = await provider.request({
+        method: "eth_sendTransaction",
+        params: [tx]
+      });
+      const receipt = await provider.request({
+        method: "eth_getTransactionReceipt",
+        params: [hash]
+      });
+      assert.notStrictEqual(receipt, null);
+      const block = await provider.request({
+        method: "eth_getBlockByNumber",
+        params: ["latest", false]
+      });
+      assert.strictEqual(
+        parseInt(block.timestamp),
+        +time / 1000 + timestampIncrement
+      );
+      await provider.disconnect();
+    });
+
+    it("uses the timestampIncrement for the first block when forking", async () => {
+      const time = new Date("2019-01-01T00:00:00.000Z");
+      const timestampIncrement = 5;
+      const fakeMainnet = await getProvider({
+        chain: { time }
+      });
+      const provider = await getProvider({
+        fork: { provider: fakeMainnet as any },
+        miner: { timestampIncrement }
+      });
+      const block = await provider.request({
```

Comment:
Yup. We do create a block immediately upon forking. We'll add an option to _not_ create a new block in the future, because it's actually really annoying. I could see Ganache v8 defaulting to this no new empty immediate block behavior.